### PR TITLE
New TaskDataset and Protonet Examples.

### LINKED ISSUE #82: Add a way to create new classes from image rotations
Currently, `l2l.vision.transforms.RandomRotation` generates additional image rotations within the same dataset. The actual functionality (following the MAML repo) is that each rotation corresponds to its own class. This means that an Omniglot task could in theory consist of labels: `[A, rot(A, 90), rot(A, 180), rot(A, 270), B]`, where `rot(A, x)` is the character A rotated by x degrees.

### PR DESCRIPTION
**Pull request by seba-1511**
### Description

Fixes #82.

Merges the `TaskGenerator` and `TaskSampler` (from Protonet PR) into a unified `TaskDataset`.

This PR includes:

* New `TaskDataset` with tests and documentation.
* New task transforms with tests and documentation.
* Port of all vision examples to use the new TaskDataset.
* Implementation of ProtoNets on mini-Imagenet, thanks to @debajyotidatta .
* New `RandomClassRotation` task transform.

### Contribution Checklist

If your contribution modifies code in the core library (not docs, tests, or examples), please fill the following checklist.

- [x] My contribution modifies code in the main library.
- [x] My modifications are tested.
- [x] My modifications are documented.

#### Optional

If you make major changes to the core library, please copy-paste the output of running `make alltests` below.

~~~shell
[PASTE HERE]
~~~

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,32 +43,33 @@ import learn2learn as l2l
...
-                                        ways=3,
-                                        classes=[0, 1, 4, 6, 8, 9],
-                                        tasks=10)
+train_tasks = l2l.data.TaskDataset(mnist,
+                                   task_transforms=[
+                                        NWays(mnist, n=3),
+                                        KShots(mnist, k=1),
+                                        LoadData(mnist),
+                                   ],
+                                   num_tasks=10)
```
Comment (debajyotidatta) [inline README.md]: This is really neat! 👯‍♂

File: examples/vision/maml_miniimagenet.py [hunk 2]
```diff
@@ -62,9 +68,39 @@ def main(
...
+                                       task_transforms=valid_transforms,
+                                       num_tasks=600)
+
+    test_transforms = [
+        l2l.data.transforms.NWays(test_dataset, ways),
+        l2l.data.transforms.KShots(test_dataset, 2*shots),
+        l2l.data.transforms.LoadData(test_dataset),
+        l2l.data.transforms.RemapLabels(test_dataset),
+        l2l.data.transforms.ConsecutiveLabels(train_dataset),
+    ]
```
Comment (debajyotidatta) [inline examples/vision/maml_miniimagenet.py]: Is it not better to just have an import of the form:

`from l2l.data.transforms import NWays, KShots, LoadData, RemapLabels, ConsecutiveLabels`

and then directly use them?
